src/modules/sensors/sensors.cpp

```diff
@@ -507,7 +509,7 @@ Sensors::adc_poll(struct sensor_combined_s &raw)
 							selected_source = b;
 
 							/* Move the selected_source to instance 0 */
-							if (_battery_pub_intance0ndx != b) {
+							if (BOARD_NUMBER_BRICKS > 1 && _battery_pub_intance0ndx != b) {
```

Comment:
Was there a compiler warning?

When BOARD_NUMBER_BRICKS == 1, b can only be 0.  The initial condition is _battery_pub_intance0ndx == 0
So  _battery_pub_intance0ndx != b is always false.